### PR TITLE
Add UserData for VM.

### PR DESCRIPTION
**Pull request by iskey**
Add UserData to `data.models` for VM.
related to #122 , #116

### CODE REVIEW

File: modules/ocl-loader/src/main/java/org/eclipse/xpanse/modules/ocl/loader/data/models/enums/UserDataType.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ * user data types for UserData.
+ */
+public enum UserDataType {
+    SHELL,
+    POWERSHELL;
+
+    private static Map<String, UserDataType> enumsMap = new HashMap<String, UserDataType>(
+            2);
+
+    static {
```
Comment (swaroopar) [inline modules/ocl-loader/src/main/java/org/eclipse/xpanse/modules/ocl/loader/data/models/enums/UserDataType.java:26]: Its better if we can avoid these maps and define enums as below.
SHELL("shell"),
POWERSHELL("powershell");

the full class will look something like this (have not tested it). This gives a better readability of enum values.
```
public enum UserDataType {

    SHELL("shell"),
    POWEpublic enum UserDataType {

    SHELL("shell"),
    POWERSHELL("powershell");

    private final String scriptType;
RSHELL("powershell");

    private final String scriptType;

    UserDataType(String scriptType) {
        this.scriptType = scriptType;
    }

    /**
     * For UserDataType serialize.
     */
    @JsonCreator
    public UserDataType getByValue(String scriptType) {
        for (UserDataType userDataType : values()) {
            if (userDataType.scriptType.equals(scriptType)) {
                return userDataType;
            }
        }
        return null;
    }

    /**
     * For UserDataType deserialize.
     */
    @JsonValue
    public String toValue() {
        return this.scriptType;
    }
}

```

**Review by swaroopar — COMMENTED** [review]
Rest looks good. Only enum class needs to changed.

File: plugins/huaweicloud/src/test/resources/huawei_test.json [hunk 2]
```diff
@@ -71,7 +71,20 @@
         "storage": [
           "$.storage[0]"
         ],
-        "publicly": true
+        "publicly": true,
```
Comment (niuzhenguo) [inline plugins/huaweicloud/src/test/resources/huawei_test.json:74]: If changing to this userdata way, public ip will be a must :(

Comment (iskey) [reply plugins/huaweicloud/src/test/resources/huawei_test.json:74]: If something must be installed from the internet, yes. If only configure something, no.

**Review by swaroopar — APPROVED** [review]

**Review by niuzhenguo — APPROVED** [review]
/lgtm

**Pull request merged**